python/python/tests/test_json.py

```diff
@@ -0,0 +1,348 @@
+                pa.field("price", pa.float64()),
+                json_field("metadata"),
+                pa.field("in_stock", pa.bool_()),
+            ]
+        ),
+    )
+
+    with tempfile.TemporaryDirectory() as tmpdir:
+        dataset_path = Path(tmpdir) / "mixed_types.lance"
+
+        # Write and read with version 2.2 (required for JSON support)
+        lance.write_dataset(table, dataset_path, data_storage_version="2.2")
+        dataset = lance.dataset(dataset_path)
+
+        # Verify all fields are preserved
+        assert len(dataset.schema) == 5
+
+        result = dataset.to_table()
+        assert result.num_rows == 2
+        assert result.column("id").to_pylist() == [1, 2]
+        assert result.column("name").to_pylist() == ["Product A", "Product B"]
+        assert result.column("in_stock").to_pylist() == [True, False]
+
+
+def test_json_append_and_overwrite():
+    """Test appending and overwriting JSON data."""
+
+    with tempfile.TemporaryDirectory() as tmpdir:
+        dataset_path = Path(tmpdir) / "append_test.lance"
+
+        # Initial data
+        initial_json = [json.dumps({"version": 1, "enabled": True})]
+        initial_ext_array = JsonArray.from_json_strings(initial_json)
+
+        initial_table = pa.table(
+            {"id": pa.array([1]), "config": initial_ext_array},
+            schema=pa.schema([pa.field("id", pa.int32()), json_field("config")]),
+        )
+
+        # Write initial data with version 2.2 (required for JSON support)
+        lance.write_dataset(initial_table, dataset_path, data_storage_version="2.2")
+
+        # Append more data
+        append_json = [json.dumps({"version": 2, "enabled": False})]
+        append_ext_array = JsonArray.from_json_strings(append_json)
+
+        append_table = pa.table(
+            {"id": pa.array([2]), "config": append_ext_array},
+            schema=pa.schema([pa.field("id", pa.int32()), json_field("config")]),
+        )
+
+        dataset = lance.dataset(dataset_path)
+        dataset = lance.write_dataset(append_table, dataset_path, mode="append")
+
+        # Verify append
+        result = dataset.to_table()
+        assert result.num_rows == 2
+        assert result.column("id").to_pylist() == [1, 2]
+
+        # Test overwrite
+        overwrite_json = [json.dumps({"version": 3, "enabled": True})]
+        overwrite_ext_array = JsonArray.from_json_strings(overwrite_json)
+
+        overwrite_table = pa.table(
+            {"id": pa.array([3]), "config": overwrite_ext_array},
+            schema=pa.schema([pa.field("id", pa.int32()), json_field("config")]),
+        )
+
+        dataset = lance.write_dataset(overwrite_table, dataset_path, mode="overwrite")
+
+        # Verify overwrite
+        result = dataset.to_table()
+        assert result.num_rows == 1
+        assert result.column("id").to_pylist() == [3]
+
+
+def test_json_filter_and_select():
+    """Test filtering and selecting with JSON columns."""
```

Comment:
Unless this test case is actually testing filtering on the JSON column I'm not sure it's warranted.  We didn't make any changes to the scanner did we?